Modules/pyexpat.c

```diff
@@ -1692,7 +1700,7 @@ MODULE_INITFUNC(void)
         }
     }
     Py_DECREF(errmod_name);
-    model_module = PyDict_GetItem(d, modelmod_name);
+    model_module = PyDict_GetItemWithError(d, modelmod_name);
```

Comment:
The model_module gets reset on line 1705, so the error from the first attempt may remain uncleared or swallowed

Reply:
Line 1705 is executed only no error was raised at this line.

Error handling in this function (as well as in many other module initialization functions) is pretty poor. Results of `PyModule_AddObject()` and derived functions are not checked, and references are leaked in case of error. But this is different issue(s).

Reply:
Then shouldn't line 1704 have ` && !PyErr_Occurred()` like line 1694 does?

Reply:
Already added.